### ISSUE TITLE
Modal backdrop doesn't show when using navbar-fixed-top

### ISSUE THREAD

**Issue by jwanglof** [angular-ui/bootstrap#2925]
```
<body>
    <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
        <div class="container-fluid"></div>
   </nav>
</body>
```

If I remove navbar-fixed-top the backdrop works as they should but if the class is present the backdrop doesn't show, but if I close the modal the backdrop flickers.

**Comment by sylouuu** [comment]
Hi,

Any decent CSS hack or something? This is *very* annoying...

Bests

**Comment by thetypebeast** [comment]
For those looking for a quick css fix: 
After examination, in my instance, `.modal-backdrop` has `z-index: 1040;` which is the same as `.modal`. I placed the modal html snippet just above `</body>`, but it seems that the html for `.modal-backdrop` is generated and inserted just before the body tag too, placing it after `.modal`. Since the z-index is the same `.modal-backdrop` gets placed on top of `.modal`.

**TL;DR**
My fix is to override the z-index of `.modal-backdrop` to a number below the z-index value of `.modal`:
```css
.modal-backdrop {
    z-index: 0;     // value is less than .modal
}
```

**Comment by wesleycho** [comment]
This appears to be fixed in `master` currently, as can be seen [here](http://plnkr.co/edit/cGCS8U?p=preview).
